app/src/main/java/zapsolutions/zap/connection/manageWalletConfigs/WalletConfigsManager.java

```diff
@@ -0,0 +1,272 @@
+            e.printStackTrace();
+        } catch (IllegalBlockSizeException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            mWalletConfigsJson = new Gson().fromJson(decrypted, WalletConfigsJson.class);
+        } catch(JsonSyntaxException e) {
+            // mWalletConfigJson is null
+        }
+    }
+
+    // used for unit tests
+    public WalletConfigsManager(String walletConfigsJson) {
+        try {
+            mWalletConfigsJson = new Gson().fromJson(walletConfigsJson, WalletConfigsJson.class);
+        } catch(JsonSyntaxException e) {
+            // mWalletConfigJson is null
+        }
+    }
+
+    public static WalletConfigsManager getInstance() {
+        if (mInstance == null) {
+            mInstance = new WalletConfigsManager();
+        }
+
+        return mInstance;
+    }
+
+
+    /**
+     * Used to determine if the provided String is a valid walletConfigs JSON.
+     *
+     * @param walletConfigsString parses as JSON
+     * @return if the JSON syntax is valid
+     */
+    public static boolean isValidJson(String walletConfigsString) {
+        try {
+            WalletConfigsJson walletConfigs = new Gson().fromJson(walletConfigsString, WalletConfigsJson.class);
+            return walletConfigs != null;
+        } catch (JsonSyntaxException ex) {
+            return false;
+        }
+    }
+
+
+    /**
+     * Checks if a wallet configuration with the given alias/name already exists.
+     *
+     * @param alias The name of the wallet
+     * @return
+     */
+    public boolean doesWalletConfigExist(String alias) {
+
+        if (alias == null) {
+            return false;
+        }
+
+        if (mWalletConfigsJson == null) {
+            return false;
+        }
+
+        return mWalletConfigsJson.getConnection(alias.toLowerCase()) != null;
+    }
+
+
+    /**
+     * Saves a wallet configuration encrypted in the default shared preferences.
+     * If a configuration exist with this alias, it will be overridden.
+     *
+     * @param alias    Name of the wallet/configuration
+     * @param type     One of the following types: remote, local
+     * @param host     The host
+     * @param port     The port
+     * @param cert     The certificate. This is optional and can be null
+     * @param macaroon The Macaroon. Encoded as base16 (hex)
+     */
+    public void saveWalletConfig(String alias, String type, String host,
+                                 int port, @Nullable String cert, String macaroon) throws IOException,
+            CertificateException, NoSuchAlgorithmException, InvalidKeyException, UnrecoverableEntryException,
+            InvalidAlgorithmParameterException, NoSuchPaddingException, NoSuchProviderException, BadPaddingException,
+            KeyStoreException, IllegalBlockSizeException {
+
+        mWalletConfigsJson = createWalletConfigJson(alias, type, host, port, cert, macaroon);
+
+        // Convert JSON object to string
```

Comment:
I guess this is the same as in line: https://github.com/LN-Zap/zap-android/pull/55/files#diff-6250c70c4d551764c2728b357ccdd562R265
So we could consolidate it